### ISSUE TITLE
Flushing slower than it could be on Windows

### ISSUE THREAD

**Issue by drako0812** [nsf/termbox-go#122]
I noticed while trying to write a small game prototype using github.com/JoelOtter/termloop which uses termbox internally, that the screen seemed to update fairly slowly.

Because of this I perused the termbox source and saw that it tries to update the least cells on the console as it can by splitting the drawing into multiple write_console_output calls.

From previous experience with the Win32 console api in C++, I generally found making one large draw performs much better than several smaller draws. This is similar to how with graphical programs you try to render things in as few draws as possible.

Now, I admit that maybe there is a reason that things aren't this way and this is a relatively minor issue for a fairly small use case.

Also, fyi the example that really exhibits this issue well is github.com/JoelOtter/termloop/_examples/image.go especially for large console window sizes. Also, termloop has another seperate issue dealing with a build-up of events when holding a key down.

**Comment by nsf** [comment]
My answer to this will be: termbox is not suited for games. The way windows update work is it updates stuff on a line by line basis. It was a torture to get this right. Doing fullscreen updates itnroduces flicker. Doing only changed cells messes up cell boundaries (some unicode box drawing characters or character backgrounds overlap on windows).

If you want to make a game - use SDL for primitive 2d drawing. Or consider one of the engines, there is a plenty of 2d game engines out there.

**Comment by drako0812** [comment]
True, using the console for games was mainly a prototyping thing for me, and it still works decently as long as the game doesn't need a high fps. I generally (for console-based games) try to limit refreshes to at max 15 fps, but most only need to refresh on input.

Also, this library still outperforms trying to do the same thing with .NET's System.Console.
